kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java

```diff
@@ -151,21 +152,25 @@ private void updateStubs() {
         span.finish();
       }
     } catch (Exception e) {
-      LOG.error("SearchMetadata failed with " + e);
+      LOG.error("SearchMetadata failed with ", e);
       List<CompletableFuture<SearchResult<LogMessage>>> emptyResultList =
           Collections.singletonList(CompletableFuture.supplyAsync(SearchResult::empty));
       return CompletableFutures.allAsList(emptyResultList);
     }
   }
 
-  public CompletableFuture<KaldbSearch.SearchResult> doSearch(KaldbSearch.SearchRequest request) {
-    CompletableFuture<List<SearchResult<LogMessage>>> searchResults = distributedSearch(request);
-
-    CompletableFuture<SearchResult<LogMessage>> aggregatedResult =
-        ((SearchResultAggregator<LogMessage>)
-                new SearchResultAggregatorImpl<>(SearchResultUtils.fromSearchRequest(request)))
-            .aggregate(searchResults);
-
-    return SearchResultUtils.toSearchResultProto(aggregatedResult);
+  public KaldbSearch.SearchResult doSearch(KaldbSearch.SearchRequest request) throws IOException {
+    try {
+      List<SearchResult<LogMessage>> searchResults =
+          distributedSearch(request).get(30, TimeUnit.SECONDS);
+      SearchResult<LogMessage> aggregatedResult =
+          ((SearchResultAggregator<LogMessage>)
+                  new SearchResultAggregatorImpl<>(SearchResultUtils.fromSearchRequest(request)))
+              .aggregate(searchResults);
+      return SearchResultUtils.toSearchResultProto(aggregatedResult);
+    } catch (Exception e) {
+      LOG.error("Distributed search failed", e);
+      throw new IOException(e);
```

Comment:
Can we throw a RuntimeException instead of a checked exception here?

Reply:
Also, can we add some logging here so we know if doSearch is really recursive as they traces indicate it to be?

Reply:
We can work on logging and other tracing additions in another issue - I think this PR we just want to remove async code

Reply:
RE - runtime exception , I left a comment on the previous conversation as it's the same